routes/course.js

```diff
@@ -67,41 +67,32 @@ router.get("/api/coursesByTerm/:plan_id", (req, res) => {
 //distribution field is also updated
 router.post("/api/courses", async (req, res) => {
   const course = req.body;
-  //console.log("course is ", course);
   await plans
     .findById(course.plan_id)
     .then((plan) => {
-      //console.log(plan);
       course.distribution_ids.forEach((id) => {
-        if (!plan.distribution_ids.includes(id)) {
+        if (!plan.distribution_ids.includes(id)) 
           errorHandler(res, 400, {
             message: "Invalid combination of plan_id and distribution_ids.",
           });
-        }
       });
     })
     .catch((err) => {
-      //console.log("here", err);
       errorHandler(res, 500, err);
     });
   courses
     .create(course)
-    .then((retrievedCourse) => {
-      retrievedCourse.distribution_ids.forEach((id) => {
-        distributions
+    .then(async (retrievedCourse) => {
+      for (let id of retrievedCourse.distribution_ids) {
+        const distribution = await distributions
```

Comment:
We made these changes to put errorHandler at the same indentation level as returnData. @mattliu-mygit could you please explain why this had to be asynchronous? thanks!

Reply:
It's because the asynchronous timeline was executing incorrectly before. This for loop, because if it's prior synchronous nature in a .then, actually ran after the returnData call below it. This threw an error and the error response was sent after the returnData success response, causing a server crash. These changes make everything correctly asynchronous. The loop executes first and then on success, the returnData is sent, only allowing for one HTTP response back.